packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -2253,8 +2259,15 @@ class InputDecoration {
   /// null.
   ///
   /// The semantic label can be replaced by providing a [semanticCounterText].
+  ///
+  /// If set to empty string and [counter] isn't specified, will display nothing
```

Comment:
If null or an empty string and [counter] isn't specified, then nothing will appear in the counter's location.